From 21ff1d170f97e52366d77b2bc7506483b48f85e3 Mon Sep 17 00:00:00 2001
From: xxh <xxh@rock-chips.com>
Date: Tue, 12 Dec 2017 17:04:39 +0800
Subject: [PATCH] net: wireless: rockchip_wlan: fix bcmdhd driver spin lock
 issue

stack backtrace:
[   95.323698] BUG: scheduling while atomic: dhd_dpc/307/0xfffffe00
[   95.323739] Modules linked in: bcmdhd
[   95.323775] CPU: 3 PID: 307 Comm: dhd_dpc Tainted: G        W       4.4.103 #13
[   95.323792] Hardware name: Rockchip RK3328 EVB (DT)
[   95.323807] Call trace:
[   95.323838] [<ffffff80080896ac>] dump_backtrace+0x0/0x1cc
[   95.323863] [<ffffff800808988c>] show_stack+0x14/0x1c
[   95.323889] [<ffffff80083830b8>] dump_stack+0x94/0xb4
[   95.323915] [<ffffff80080c58c0>] __schedule_bug+0x40/0x54
[   95.323941] [<ffffff8008b0c250>] __schedule+0x8c/0x6f0
[   95.323964] [<ffffff8008b0c934>] schedule+0x80/0xa0
[   95.323991] [<ffffff8008b0f258>] schedule_timeout+0x3c/0x260
[   95.324015] [<ffffff8008b0d48c>] wait_for_common+0x130/0x168
[   95.324040] [<ffffff8008b0d4d8>] wait_for_completion+0x14/0x1c
[   95.324066] [<ffffff800876947c>] mmc_wait_for_req+0x84/0x160
[   95.324091] [<ffffff8008774518>] mmc_io_rw_extended+0x26c/0x2ec
[   95.324117] [<ffffff8008775958>] sdio_io_rw_ext_helper+0x14c/0x1a0
[   95.324141] [<ffffff8008775b50>] sdio_readsb+0x1c/0x24
[   95.325601] [<ffffff8000b5d480>] sdioh_buffer_tofrom_bus+0xc0/0x238 [bcmdhd]
[   95.327078] [<ffffff8000b5e910>] sdioh_request_buffer+0x1c0/0x360 [bcmdhd]
[   95.328529] [<ffffff8000b5c1c8>] bcmsdh_recv_buf+0x88/0xc0 [bcmdhd]
[   95.329991] [<ffffff8000b6412c>] dhd_bcmsdh_recv_buf.constprop.25+0x64/0x98 [bcmdhd]
[   95.331444] [<ffffff8000b66fb8>] dhdsdio_readframes+0x908/0x14e0 [bcmdhd]
[   95.332916] [<ffffff8000b6bc20>] dhd_bus_dpc+0x6c0/0xb80 [bcmdhd]
[   95.334004] [<ffffff8000b35cd8>] dhd_dpc_thread+0x128/0x1a0 [bcmdhd]
[   95.334023] [<ffffff80080bc6e4>] kthread+0xe8/0xf0
[   95.334035] [<ffffff80080828d0>] ret_from_fork+0x10/0x40

Change-Id: I56c837ca08e946970b873e72cc87150592d5eaad
Signed-off-by: Xu Xuehui <xxh@rock-chips.com>
---
 .../net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c  | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c
index 4df7ea6d84e9..b77ecac9ba6f 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c
@@ -13674,19 +13674,11 @@ dhd_os_sdunlock_txq(dhd_pub_t *pub)
 void
 dhd_os_sdlock_rxq(dhd_pub_t *pub)
 {
-	dhd_info_t *dhd;
-
-	dhd = (dhd_info_t *)(pub->info);
-	spin_lock_bh(&dhd->rxqlock);
 }
 
 void
 dhd_os_sdunlock_rxq(dhd_pub_t *pub)
 {
-	dhd_info_t *dhd;
-
-	dhd = (dhd_info_t *)(pub->info);
-	spin_unlock_bh(&dhd->rxqlock);
 }
 
 static void
-- 
2.35.3

